### PR TITLE
Add an explicit test for loading the yajl2_c extension

### PR DESCRIPTION
**Pull request by mgorny**
Following the discussion in #98, add an explicit test that verifies that the yajl2_c extension can be loaded successfully if it was built. If the extension module is not present, the test is skipped.

### CODE REVIEW

File: test/test_misc.py [hunk 1]
```diff
@@ -16,6 +17,13 @@ def test_common_number_is_deprecated(self):
         self.assertEqual(len(warns), 1)
         self.assertEqual(DeprecationWarning, warns[0].category)
 
+    def test_yajl2_c_loadable(self):
+        spec = importlib.util.find_spec("ijson.backends._yajl2")
```
Comment (rtobar) [inline test/test_misc.py:21]: I know this is completely out of date, but could you skip the test for python 2.7, since this functionality was only introduced in 3.4? I'm definitely planning to stop supporting 2.7 formally soon, but that entails some proper cleanups here and there that will take some time. Until then I'd rather ensure all tests still pass there too.

File: test/test_misc.py [hunk 2]
```diff
@@ -94,4 +102,4 @@ def test_rich_items(self):
     def test_rich_kvitems(self):
         self._assert_entry_point(JSON_KVITEMS, self.parse, self.kvitems, 'docs.item')
 
-generate_backend_specific_tests(globals(), 'MainEntryPoints', '', MainEntryPoints)
\ No newline at end of file
+generate_backend_specific_tests(globals(), 'MainEntryPoints', '', MainEntryPoints)
```
Comment (rtobar) [inline test/test_misc.py:105]: Let's not have this unnecessary change here, thanks

**Review by rtobar — COMMENTED** [review]
@mgorny thanks again for this PR! I've left two minor points to address, but otherwise it looks great. I've also pushed to the `master` branch some changes to ensure our GitHub Actions are run on PRs (they were not, which I wasn't aware of). Ideally if you could rebase on top of the latest `master` and force-push your commit with these comments addressed that'd be great, as I'd rather keep a linear history in the repo; otherwise I can do that too at merge time.

File: test/test_misc.py (hunk 1 above)
Comment (mgorny) [reply test/test_misc.py:21]: Done. Sorry, it didn't even occur to me to look for Python 2 compatibility.

File: test/test_misc.py (hunk 2 above)
Comment (mgorny) [reply test/test_misc.py]: I think I've managed that. Could you please add trailing newlines to all files, though? Some editors just don't let you save files without that.

**Review by rtobar — COMMENTED** [review]
almost there!

File: test/test_misc.py [hunk 3]
```diff
@@ -16,6 +17,15 @@ def test_common_number_is_deprecated(self):
         self.assertEqual(len(warns), 1)
         self.assertEqual(DeprecationWarning, warns[0].category)
 
+    def test_yajl2_c_loadable(self):
+        if not compat.IS_PY2:
```
[the remark this answers is not in the document]
Comment (mgorny) [reply test/test_misc.py:21]: Right, sorry.

**Comment by rtobar** [comment]
Thanks again @mgorny for the new test!
